### ISSUE TITLE
HASS v0.110 Component Rename

### ISSUE THREAD

**Issue by ssilence5** [twrecked/hass-aarlo#249]
Thanks to Grant on the Home-assistant community, it looks like there is a change coming to the naming of some of the component types in HASS v0.110. Here is the full warning log that was presented:

```2020-05-15 19:33:36 WARNING (MainThread) [homeassistant.components.switch] SwitchDevice is deprecated, modify AarloSwitch to extend SwitchEntity
2020-05-15 19:33:36 WARNING (MainThread) [homeassistant.components.switch] SwitchDevice is deprecated, modify AarloSirenBaseSwitch to extend SwitchEntity
2020-05-15 19:33:36 WARNING (MainThread) [homeassistant.components.switch] SwitchDevice is deprecated, modify AarloSirenSwitch to extend SwitchEntity
2020-05-15 19:33:36 WARNING (MainThread) [homeassistant.components.switch] SwitchDevice is deprecated, modify AarloAllSirensSwitch to extend SwitchEntity
2020-05-15 19:33:36 WARNING (MainThread) [homeassistant.components.switch] SwitchDevice is deprecated, modify AarloSnapshotSwitch to extend SwitchEntity
2020-05-15 19:33:36 WARNING (MainThread) [homeassistant.components.alarm_control_panel] AlarmControlPanel is deprecated, modify ArloBaseStation to extend AlarmControlPanelEntity
2020-05-15 19:33:36 WARNING (MainThread) [homeassistant.components.binary_sensor] BinarySensorDevice is deprecated, modify ArloBinarySensor to extend BinarySensorEntity
```

For now it looks like the old names will work, but you get a warning. Below are the links to the commits in HA Core that are affecting this:

- SwitchDevice to SwitchEntity: https://github.com/home-assistant/core/commit/066e921a8b6f5264de4a93f604aed789c9730b73#diff-a848334e018338b417be2506f5f99f40
- AlarmControlPanel to AlarmControlPanelEntity: https://github.com/home-assistant/core/commit/6f7f5b4034bc55246a8fa170dd330b1edec9ea57#diff-fb6f4f6b0a5670a75529a7e7cd0047e9
- BinarySensorDevice to BinarySensorEntity: https://github.com/home-assistant/core/commit/b022e08db9b722af39fe34ebd543d81be8e1744d#diff-3d99deefc08c356e814e2f36025af012

**Comment by twrecked** [comment]
Thanks for the heads up. It looks like a straight swap so I'll push a fix when 110 is the main release.

**Comment by ssilence5** [comment]
Yeah, I was going to make the mod and issue a pull request, but wasn't sure how you wanted to handle this to make things backwards compatible. Wasn't sure if you wanted to somehow detect the HASS version and use the old name if pre v0.110 and the new name after (not even sure if you can do that).

**Comment by dlo747** [comment]
I opened up #255 however just noticed you already had mentioned it here

**Comment by twrecked** [comment]
@ssilence5 I was just going to wait until 0.110.0 came out. A lazy updating strategy!

Fixed with PR #256 and in release 0.6.19.
